javascript_nodejs/81.facebook-events/bot.js

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { ActivityTypes, MessageFactory } = require('botbuilder');
+const { ChoicePrompt, DialogSet } = require('botbuilder-dialogs');
+
+const DIALOG_STATE_PROPERTY = 'dialogState';
+const USER_PROFILE_PROPERTY = 'user';
+const SAMPLE_PROMPT = 'sample_prompt';
+
+/**
+ * A simple bot that responds handles and shows Facebook events.
+ */
+class FacebookEventsBot {
+    /**
+     * 
+     * @param {ConversationState} conversationState A ConversationState object used to store the dialog state.
+     * @param {UserState} userState A UserState object used to store values specific to the user.
+     */
+    constructor (conversationState, userState) {
+
+        this.conversationState = conversationState;
```

Comment:
Comment above indicating why we need conversationState in the constructor, i.e. saving state and creating propertyaccessors.